src/3d/shaders/lines.geom

```diff
@@ -109,16 +118,28 @@ void main( void )
             VertexOut.mTexCoord = vec2( 0, 0 );
             VertexOut.mColor = VertexIn[1].mColor;
             gl_Position = vec4( ( p1 + THICKNESS * n1 ) / WIN_SCALE, p1z, 1.0 );
+#ifdef CLIPPING
+            worldPosition = vec3(inverseViewProjectionMatrix * (gl_Position * px1.w));
```

Comment:
I don't get the multiplication by `px1.w` (or `px2.w`) here and below. If we're dealing with homogeneous coordinates here, I can multiply by any constant and still refer to the same point... or not?

Reply:
I don't have an explanation but if remove the multiplication by `px1.w`, the result is completely wrong while it works as expected with it.

Reply:
My explanation is that it works somehow just accidentally, because px1 is a point that's not far from gl_Position.

I think I understand what's going on: gl_Position is in homogeneous coordinates in clip space, the inverse view-projection matrix transforms it to world space, still in homogeneous coordinates. And setClipDistance() works in world space, but it only takes `vec3`, so the world space coordinates must have w=1. So I guess the proper fix would be:
```
vec4 worldPosition = inverseViewProjectionMatrix * gl_Position;
worldPosition /= worldPosition.w
```
Alternatively, you could modify setClipDistance() to take vec4 rather than vec3.

Reply:
Thanks for the investigation and the explanation. I have to admit that I have very little knowledge regarding shader code.